pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go

```diff
@@ -190,7 +190,7 @@ func (r *ReconcileNetworkAddonsConfig) Reconcile(request reconcile.Request) (rec
 			// Request object not found, could have been deleted after reconcile request.
 			// Reset list of tracked objects.
 			// TODO: This can be dropped once we implement a finalizer waiting for all components to be removed
-			r.trackDeployedObjects([]*unstructured.Unstructured{})
+			r.trackDeployedObjects([]*unstructured.Unstructured{}, 0)
```

Comment:
This may introduce a bug. When we remove a NAC, this will be set tp empty list and 0. Let's say somebody then creates a new NAC. It will get generation 0 and couple of containers assigned. We can get into the same race again. Can we use negative number for the intial setting?

Or maybe better, can we introduce r.stopTracking. That would speak for itself.

Reply:
added method `stopTrackingObjects` and used negative value to reset the generation.